### PR TITLE
Remove TodoGenerator from Api template

### PR DESCRIPTION
**Pull request by DamianEdwards**
Removes the `TodoGenerator` from the Api project template and simplifies it to a single .cs file:

```csharp
var builder = WebApplication.CreateSlimBuilder(args);

var app = builder.Build();

var sampleTodos = new Todo[] {
    new(1, "Walk the dog"),
    new(2, "Do the dishes", DateOnly.FromDateTime(DateTime.Now)),
    new(3, "Do the laundry", DateOnly.FromDateTime(DateTime.Now.AddDays(1))),
    new(4, "Clean the bathroom"),
    new(5, "Clean the car", DateOnly.FromDateTime(DateTime.Now.AddDays(2)))
};

var todosApi = app.MapGroup("/todos");
todosApi.MapGet("/", () => sampleTodos);
todosApi.MapGet("/{id}", (int id) =>
    sampleTodos.FirstOrDefault(a => a.Id == id) is { } todo
        ? Results.Ok(todo)
        : Results.NotFound());

app.Run();

public record Todo(int Id, string? Title, DateOnly? DueBy = null, bool IsComplete = false);
```

and with `-aot` parameter:

```csharp
using System.Text.Json.Serialization;

var builder = WebApplication.CreateSlimBuilder(args);

builder.Services.ConfigureHttpJsonOptions(options =>
{
    options.SerializerOptions.TypeInfoResolverChain.Insert(0, AppJsonSerializerContext.Default);
});

var app = builder.Build();

var sampleTodos = new Todo[] {
    new(1, "Walk the dog"),
    new(2, "Do the dishes", DateOnly.FromDateTime(DateTime.Now)),
    new(3, "Do the laundry", DateOnly.FromDateTime(DateTime.Now.AddDays(1))),
    new(4, "Clean the bathroom"),
    new(5, "Clean the car", DateOnly.FromDateTime(DateTime.Now.AddDays(2)))
};

var todosApi = app.MapGroup("/todos");
todosApi.MapGet("/", () => sampleTodos);
todosApi.MapGet("/{id}", (int id) =>
    sampleTodos.FirstOrDefault(a => a.Id == id) is { } todo
        ? Results.Ok(todo)
        : Results.NotFound());

app.Run();

public record Todo(int Id, string? Title, DateOnly? DueBy = null, bool IsComplete = false);

[JsonSerializable(typeof(Todo[]))]
internal partial class AppJsonSerializerContext : JsonSerializerContext
{

}
```

### CODE REVIEW

File: src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/Program.cs [hunk 1]
```diff
@@ -25,6 +30,8 @@
 
 app.Run();
 
+public record Todo(int id, string? title, DateOnly? dueBy = null, bool isComplete = false);
```
Comment (davidfowl) [inline src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/Program.cs:33]: These should be pascal case as they are properties

**Review by davidfowl — APPROVED** [review]

**Comment by mitchdenny** [comment]
Do you think it is worth changing from CreateSlimBuilder to CreateBuilder  based on the AOT switch. We've seen some confusion around people trying to use the API template whilst targetting IIS.

**Comment by DamianEdwards** [comment]
> Do you think it is worth changing from CreateSlimBuilder to CreateBuilder based on the AOT switch. We've seen some confusion around people trying to use the API template whilst targetting IIS.

Yeah maybe. There's also some new thoughts about potentially combining the webapi and api project templates after all, rather than introduce a new template altogether, but we can pursue that separately from this PR.

**Comment by cakescience** [comment]
> There's also some new thoughts about potentially combining the webapi and api project templates after all, rather than introduce a new template altogether

It's true that it's quite confusing and the difference is not so obvious currently

**Comment by cakescience** [comment]
> There's also some new thoughts about potentially combining the webapi and api project templates after all, rather than introduce a new template altogether

It's true that it's quite confusing and the difference between the two templates is not so obvious currently.

**Review by eiriktsarpalis — APPROVED** [review]

File: src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/Program.cs [hunk 2]
```diff
@@ -1,7 +1,6 @@
 #if (NativeAot)
 using System.Text.Json.Serialization;
 #endif
-using Company.ApiApplication1;
```
Comment (eerhardt) [inline src/ProjectTemplates/Web.ProjectTemplates/content/Api-CSharp/Program.cs:4]: Can you move this blank line to inside the `#if (NativeAot)` section? That way the file doesn't start with an empty line when you don't pass `--aot`.

**Review by eerhardt — APPROVED** [review]

**Comment by DamianEdwards** [comment]
The failed test seems unrelated to this PR? @BrennanConroy

**Comment by BrennanConroy** [comment]
Your last build is probably from before https://github.com/dotnet/aspnetcore/commit/33a8622aab3b08c338c2df15da5a0079078587a1 went in.

**Comment by eerhardt** [comment]
@DamianEdwards - are you going to update the Stage1 benchmark was well? That way it can stay in sync. It already removed the generator part, but it would be nice to have the code the same. (for example, the `record` usage).

**Comment by DamianEdwards** [comment]
@eerhardt see https://github.com/aspnet/Benchmarks/pull/1877

**Pull request merged**